src/main/java/se/liu/ida/hefquin/engine/queryplan/logical/LogicalPlanCounter.java

```diff
@@ -0,0 +1,14 @@
+package se.liu.ida.hefquin.engine.queryplan.logical;
+
+public class LogicalPlanCounter {
+	private static int counter = 1;  
+    protected int operatorID;  
```

Comment:
Make this `final`, so it cannot accidentally be changed anywhere.

And I think it is fine to name this member variable simply `id`.